host.go

```diff
@@ -322,16 +322,19 @@ func (h *Host) generateDockerConfig(dockerPort int, caCertPath string, serverKey
 		daemonOpts    string
 		daemonOptsCfg string
 		daemonCfg     string
+		swarmLabels   = []string{}
 	)
 
-	// TODO @ehazlett: template?
+	if h.SwarmHost != "" {
+		swarmLabels = append(swarmLabels, fmt.Sprintf("--label=provider=%s", h.Driver.DriverName()))
+	}
+
 	defaultDaemonOpts := fmt.Sprintf(`--tlsverify \
 --tlscacert=%s \
 --tlskey=%s \
---tlscert=%s`, caCertPath, serverKeyPath, serverCertPath)
+--tlscert=%s %s`, caCertPath, serverKeyPath, serverCertPath, strings.Join(swarmLabels, " "))
```

Comment:
sure -- the reason i went multi-line is i was anticipating more engine options being added.  if we go with cloudinit it won't be there anyway.  i can update it - it should evaluate the same.